src/main/java/com/example/travelservice/services/WeatherServiceImpl.java

```diff
@@ -2,24 +2,34 @@
 
 import com.example.travelservice.dto.WeatherDto;
 import com.example.travelservice.dto.WeatherRequest;
+import com.example.travelservice.dtoconverters.WeatherResponse;
 import com.example.travelservice.integration.WeatherChannels;
 import com.example.travelservice.integration.WeatherFromApiService;
 import com.example.travelservice.model.Trip;
-import lombok.RequiredArgsConstructor;
+import com.example.travelservice.springevents.TripSpringEvent;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.cloud.stream.annotation.StreamListener;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.annotation.Lazy;
 import org.springframework.messaging.support.MessageBuilder;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
 @Service
-@RequiredArgsConstructor
 @Slf4j
 public class WeatherServiceImpl implements WeatherService {
 
     private final WeatherFromApiService weatherFromApiService;
     private final WeatherChannels weatherChannels;
+    @Autowired
+    private ApplicationEventPublisher applicationEventPublisher;
```

Comment:
make it final and inject via constructor as other beans here

Reply:
make sense

Reply:
resolved